pkg/components/prometheus-operator/component_test.go

```diff
@@ -17,46 +17,77 @@ package prometheus
 import (
 	"testing"
 
-	"github.com/hashicorp/hcl/v2"
-
 	"github.com/kinvolk/lokomotive/pkg/components/util"
 )
 
-func TestEmptyConfig(t *testing.T) {
-	c := newComponent()
-	emptyConfig := hcl.EmptyBody()
-	evalContext := hcl.EvalContext{}
-	diagnostics := c.LoadConfig(&emptyConfig, &evalContext)
-	if !diagnostics.HasErrors() {
-		t.Fatalf("Empty config should return error")
-	}
-}
-
+// nolint:funlen
 func TestRenderManifest(t *testing.T) {
-	configHCL := `
+	tests := []struct {
+		desc    string
+		hcl     string
+		wantErr bool
+	}{
+		{
+			desc: "essential values only",
+			hcl: `
 component "prometheus-operator" {
-  grafana_admin_password = "foo"
-  namespace              = "monitoring"
-}
-`
+  grafana {
+    admin_password = "foobar"
+  }
+  namespace = "monitoring"
+}`,
+		},
+		{
+			desc: "no values",
+			hcl:  `component "prometheus-operator" {}`,
+		},
+		{
+			desc: "ingress and host given",
+			hcl: `
+component "prometheus-operator" {
+  grafana {
+    ingress {
+	  host = "foobar"
+	}
+  }
+}`,
+		},
+		{
+			desc: "ingress and no host given",
+			hcl: `
+component "prometheus-operator" {
+  grafana {
+    ingress {}
+  }
+}`,
+			wantErr: true,
+		},
+	}
 
-	component := newComponent()
+	for _, tc := range tests {
+		b, d := util.GetComponentBody(tc.hcl, name)
+		if d != nil {
+			t.Errorf("%s - Error getting component body: %v", tc.desc, d)
+		}
 
-	body, diagnostics := util.GetComponentBody(configHCL, name)
-	if diagnostics != nil {
-		t.Fatalf("Error getting component body: %v", diagnostics)
-	}
+		c := newComponent()
+		d = c.LoadConfig(b, nil)
 
-	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
-	if diagnostics.HasErrors() {
-		t.Fatalf("Valid config should not return error, got: %s", diagnostics)
-	}
+		if !tc.wantErr && d.HasErrors() {
+			t.Errorf("%s - Valid config should not return error, got: %s", tc.desc, d)
+		}
 
-	m, err := component.RenderManifests()
-	if err != nil {
-		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
-	}
-	if len(m) <= 0 {
-		t.Fatalf("Rendered manifests shouldn't be empty")
+		if tc.wantErr && !d.HasErrors() {
+			t.Errorf("%s - Wrong config should have returned an error", tc.desc)
+		}
+
+		m, err := c.RenderManifests()
```

Comment:
If I want error, meaning loading the HCL failed, this should not be executed, or?

Reply:
By want error it means that the test is expecting an error.

Reply:
yes, sorry. I mean, if we fail to load the HCL, the component will be in the possibly undefined state, so we should not try to render it then.

Reply:
Yes changed that now it is using t.Run so now I can return using `t.Fatal`.